plugins/regex_remap/regex_remap.cc

```diff
@@ -363,19 +372,19 @@ RemapRegex::initialize(const std::string &reg, const std::string &sub, const std
           break;
         default:
           TSError("[%s] configuration variable '%s' is of an unsupported type", PLUGIN_NAME, opt_name.c_str());
-          delete cur;
           return false;
         }
         TSDebug(PLUGIN_NAME, "Overridable config %s=%s", opt_name.c_str(), opt_val.c_str());
         cur->key  = key;
         cur->type = type;
         cur->next = nullptr;
+        auto tmp  = cur.get();
```

Comment:
Wondering what the point of `tmp` is, since it's only used once. It's a net increase in code for no apparent benefit.